src/test/java/com/team701/buddymatcher/controllers/pairing/PairingControllerIntegrationTest.java

```diff
@@ -136,20 +136,24 @@ void requestBuddyMatchFromMultipleCoursesWithMultipleStudents() throws Exception
                 .andExpect(jsonPath("$[2].pairingEnabled").value(true));
     }
 
+    /**
+     * Will also check that buddies are correctly prioritised, with more shared courses appearing first
+     * @throws Exception
+     */
     @Test
     void requestBuddyMatchFromCoursesWithSameStudentInBothCourses() throws Exception {
         mvc.perform(post("/api/pairing/matchBuddy")
                         .content("{\"courseIds\":[1,4]}")
                         .contentType(MediaType.APPLICATION_JSON)
                         .sessionAttrs(Collections.singletonMap("UserId", 4)))
                 .andExpect(status().isOk())
-                .andExpect(jsonPath("$[0].id").value(2))
-                .andExpect(jsonPath("$[0].name").value("Green Dinosaur"))
-                .andExpect(jsonPath("$[0].email").value("green.dinosaur@gmail.com"))
+                .andExpect(jsonPath("$[0].id").value(3))
+                .andExpect(jsonPath("$[0].name").value("Hiruna Smith"))
+                .andExpect(jsonPath("$[0].email").value("hiruna.smith@gmail.com"))
                 .andExpect(jsonPath("$[0].pairingEnabled").value(true))
-                .andExpect(jsonPath("$[1].id").value(3))
-                .andExpect(jsonPath("$[1].name").value("Hiruna Smith"))
-                .andExpect(jsonPath("$[1].email").value("hiruna.smith@gmail.com"))
+                .andExpect(jsonPath("$[1].id").value(2))
+                .andExpect(jsonPath("$[1].name").value("Green Dinosaur"))
+                .andExpect(jsonPath("$[1].email").value("green.dinosaur@gmail.com"))
                 .andExpect(jsonPath("$[1].pairingEnabled").value(true));
     }
```

Comment:
Are you able to include test cases for when 400 and 404 are returned, please?

Reply:
I think that those cases are already tested in invalidRequestBuddyMatchFromCourseWithOneBuddy?

Reply:
Sorry, I remember now.